.github/workflows/upgrade.yml

```diff
@@ -213,9 +213,19 @@ jobs:
             DEPLOY_FLAGS='--trust'
           fi
           
-          juju deploy ${CHARM_${{ matrix.cloud }}} $DEPLOY_FLAGS
+          CURRENT_CHARM=${CHARM_${{ matrix.cloud }}}
           
-          juju wait-for application ${CHARM_${{ matrix.cloud }}}
+          juju deploy ${CURRENT_CHARM} $DEPLOY_FLAGS
+                    
+          attempt=0          
+          until juju status --format=json | jq -S ".applications | select((.[\"${CURRENT_CHARM}\"] | .[\"application-status\"] | .current == \"active\" ))" | grep "${CURRENT_CHARM}"; do
```

Comment:
Since this is inside a GitHub action, I don't see how you could easily use the `wait_for` from the testing harness. Hence I think it's fine to just put the custom loop here. Though you could check if there's a script in the `.github` folder that could help you, or potentially add one there.

Long term we want this upgrade workflow to be a standard Bash test suite like the others, we don't really have the setup for that yet though.

Reply:
We do this kind of thing in a lot of places in our actions. It would be nice to have some kind of centralised tool/script, but not important enough to matter block